static/conformance-view.js

```diff
@@ -0,0 +1,314 @@
+        if (editorId == this.editorId) {
+            _.defer(function (self) {
+                self.container.close();
+            }, this);
+        }
+    };
+
+    Conformance.prototype.onCompileResponse = function (request, result, cached) {
+        var allText = _.pluck((result.stdout || []).concat(result.stderr || []), 'text').join("\n");
+        var failed = result.code !== 0;
+        var warns = !failed && !!allText;
+        var status = {
+            text: allText,
+            code: failed ? 3 : (warns ? 2 : 1)
+        };
+        this.handleStatusIcon(this.selectorList.find('[data-cv="' + request.extras.cv + '"] .status'), status);
+        this.saveState();
+    };
+
+    Conformance.prototype.compileAll = function() {
+        Compiler.Compiler.prototype.expand(this.source).then(_.bind(function (expanded) {
+            this.currentExpandedSource = expanded;
+            // And trun off every icon
+            var compileCount = 0;
+            _.each(this.selectorList.children(), _.bind(function(child) {
+                var picker = $(child).find('.compiler-picker');
+                if (picker && compileCount < this.maxCompilations) {
+                    compileCount++;
+                    this.compile(picker);
+                }
+            }, this));
+            // Save the state
+            //this.saveState();
+        }, this));
+    };
+
+    Conformance.prototype.handleToolbarUI = function() {
+        var compilerCount = this.selectorList.children().length;
+        this.status.allowCompile = compilerCount > 0;
+        this.status.allowAdd = compilerCount < this.maxCompilations;
+
+        this.compileButton.attr("disabled", !this.status.allowCompile);
+        this.addCompilerButton.attr("disabled", !this.status.allowAdd);
+
+        this.setTitle(compilerCount);
+    };
+
+    Conformance.prototype.handleStatusIcon = function(element, status) {
+        if (!element) return;
+        element.attr("class", "status glyphicon glyphicon-" + (status.code === 3 ? "remove-sign" : (status.code === 2 ? "info-sign" : "ok-sign")))
+            .css("visibility", status.code === 0 ? "hidden" : "visible")
+            .css("color",  status.code === 3 ? "red" : (status.code === 2 ? "yellow" : "green"))
+            .attr("title", status.text)
+            .attr("aria-label", status.code === 3 ? "Compilation failed!" : (status.code === 2 ? "Compiled with warnings" : "Compiled without warnings"))
+            .attr("data-status", status.code);
+    };
+
+    Conformance.prototype.currentState = function () {
+        var state = {
+            editorid: this.editorId,
+            source: this.source,
+            compilers: []
+        };
+        _.each(this.selectorList.children(), _.bind(function(child) {
+            var status = $(child).find('.status');
+            state.compilers.push({
+                // Have we been compiled? If so, what is the result?
+                status: {
```

Comment:
Ah understood. I'm not sure I like that so much: a lot of the power of CE comes from its "compile as you type feel". Let me try it out before you change that though!